modules/custom/az_paragraphs/src/Plugin/Field/FieldFormatter/AZBackgroundMediaFormatter.php

```diff
@@ -0,0 +1,627 @@
+
+    $settings = $this->getAllSettings($items);
+
+    $element = [];
+
+    $full_width = '';
+    $marquee_style = $settings['style'];
+    if (!empty($settings['full_width'])) {
+      $full_width = $settings['full_width'];
+    }
+    $media_items = $this->getEntitiesToView($items, $langcode);
+
+    // Early opt-out if the field is empty.
+    if (empty($media_items)) {
+      return $element;
+    }
+
+    $paragraph = $items->getEntity();
+
+    // Prepare token data in bg image CSS selector.
+    $token_data = [
+      'user' => $this->currentUser,
+      $items->getEntity()->getEntityTypeId() => $items->getEntity(),
+    ];
+
+    $settings['css_settings']['selector'] = $this->token->replace(
+      $settings['css_settings']['selector'],
+      $token_data
+    );
+    $settings['css_settings'] = str_replace(['_'], '-', $settings['css_settings']);
+
+    /** @var \Drupal\media\MediaInterface[] $media_items */
+    foreach ($media_items as $delta => $media) {
+      $element['#media_type'] = $media->bundle();
+
+      switch ($media->bundle()) {
+        case 'az_remote_video':
+          $element[$delta] = $this->remoteVideo($settings, $media);
+          break;
+
+        case 'az_image':
+          $element[$delta] = $this->image($settings, $media);
+          break;
+
+        default:
+          return $element;
+      }
+
+    }
+
+    return $element;
+  }
+
+  /**
+   * Get the possible responsive image styles.
+   *
+   * @param bool $withNone
+   *   True to include the 'None' option, false otherwise.
+   *
+   * @return array
+   *   The select options.
+   */
+  protected function getResponsiveImageStyles($withNone = FALSE) {
+
+    $styles = $this->entityTypeManager->getStorage('responsive_image_style')->loadMultiple();
+    $options = [];
+
+    if ($withNone && empty($styles)) {
+      $options[''] = t('- Defined None -');
+    }
+
+    foreach ($styles as $name => $style) {
+      $options[$name] = $style->label();
+    }
+
+    return $options;
+  }
+
+  /**
+   * Get the URI for the media item.
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   The media item.
+   *
+   * @return \Drupal\Core\Uri|null
+   *   The URI object for the media item's thumbnail image.
+   */
+  protected function getMediaThumbFile(MediaInterface $media) {
+
+    $uri = NULL;
+    $file = $media->getSource();
+    $uri = $file->getMetadata($media, 'thumbnail_uri');
+    /** @var \Drupal\file\FileInterface[] $files */
+    $files = $this->entityTypeManager
+      ->getStorage('file')
+      ->loadByProperties(['uri' => $uri]);
+    /** @var \Drupal\file\FileInterface|null $file */
+    $file = reset($files) ?: NULL;
+
+    return $file;
+  }
+
+  /**
+   * Get the paragraph instance settings.
+   *
+   * @return array
+   *   The paragraph behavior settings.
+   */
+  protected function getParagraphSettings(FieldItemListInterface $items) {
+    $paragraph_settings = NULL;
```

Comment:
Nitpick:
```suggestion
    $paragraph_settings = [];
```

Reply:
Makes sense, since this returns an array https://github.com/az-digital/az_quickstart/pull/1059#discussion_r739550459